web/app/controllers/search.js

```diff
@@ -0,0 +1,196 @@
+     * We need to tie the jobType with the value of the jobtype selection input whenever the checkbox is checked.
+     */
+    watchJobCheck: function () {
+        if (!this.get("isJobTypeChecked")) {
+            this.set("jobType", null);
+        } else {
+            this.set("jobType", this.get("model.searchOptions.jobcategory").get('firstObject').jobtypes.get('firstObject').name);
+        }
+    }.observes('isJobTypeChecked'),
+
+    /**
+     * Watches the isFinishDateChecked boolean flag. This flag is true when the checkbox for FinishDate is ticked.
+     * We need to tie the finishTimeBegin and finishTimeEnd  with the value of the jobtype selection input whenever the checkbox is checked.
+     */
+    watchFinishTimeCheck: function () {
+        this.set("finishTimeBegin", null);
+        this.set("finishTimeEnd", null);
+        this.set("finishTimeBeginValue", null);
+        this.set("finishTimeEndValue", null);
+    }.observes('isFinishDateChecked'),
+
+    /**
+     * Watches the isSeverityChecked boolean flag. This flag is true when the checkbox for Severity is ticked.
+     * We need to tie the severity and analysis with the value of the severity and analysis selection input whenever the checkbox is checked.
+     */
+    watchSeverityCheck: function () {
+        if (!this.get("isSeverityChecked")) {
+            this.set("analysis", null);
+            this.set("severity", null);
+        } else {
+            this.set("severity", this.get("model.searchOptions.severities").get('firstObject').value);
+        }
+    }.observes("isSeverityChecked"),
+
+    /**
+     * Actions
+     **/
+    actions: {
+
+        /**
+         * Actions for select inputs
+         */
+
+        selectHeuristic(heuristic) {
+            this.set("analysis", heuristic);
+        },
+        selectSeverity(severity) {
+            this.set("severity", severity);
+        },
+        selectJobType(jobType) {
+            this.set("jobType", jobType);
+        },
+
+        /**
+         * loads the page
+         */
+        loadPage (page) {
+            var _this = this;
+            this.set("offset", this.get("entriesPerPage") * (page - 1));
+            this.set("limit", this.get("entriesPerPage"));
+            var newsummaries = this.store.queryRecord('search-result', {
+                'username': this.username,
+                'queue-name': this.queueName,
+                'job-type': this.jobType,
+                'severity': this.severity,
+                'analysis': this.analysis,
+                'finished-time-begin': moment(this.get('finishTimeBegin')).valueOf(),
+                'finished-time-end': moment(this.get('finishTimeEnd')).valueOf(),
+                'type': this.type,
+                'offset': this.offset,
+                'limit': this.limit
+            });
+
+            /**
+             * update model after fetching the searched data
+             */
+            newsummaries.then(function() {
```

Comment:
You can use arrow functions to maintain the context inside the function, something like this:
`newsummaries.then(() => {...}`
More information about arrow functions in [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)